metadata.go

```diff
@@ -0,0 +1,18 @@
+package iavl
+
+import (
+	"google.golang.org/protobuf/proto"
+)
+
+// Marshal returns Protobuf marshaled bytes of a VersionMetadata object.
+func (vm *VersionMetadata) Marshal() ([]byte, error) {
+	return proto.Marshal(vm)
+}
+
+// VersionMetadata attempts to decode a VersionMetadata object from Protobuf
+// marshaled bytes.
+func UnmarshalVersionMetadata(bz []byte) (*VersionMetadata, error) {
+	vm := &VersionMetadata{}
+	err := proto.Unmarshal(bz, vm)
+	return vm, err
+}
```

Comment:
We may not want to expose these as part of our public API. Can we make them private?

Reply:
Sure, but there's no harm in keeping them public. A client has no way of interacting with the persistence of metadata.

Reply:
Public APIs are subject to backwards compatibility guarantees, keeping them internal gives us more flexibility.

Reply:
This API would never change and is that a guarantee that we hold in the IAVL lib? We do this all the time in the SDK and Tendermint, where Tendermint is a worse offender because even point releases break APIs.

Reply:
Furthermore, what if an upstream client wants to serialize and store metadata for whatever reason?

Reply:
Yeah, that's sort of the thing - if we make the marshaling public then we're committing to a certain encoding, making it harder to change if we ever decide to. If we feel like that's necessary then I'm not totally opposed to it, I just prefer keeping implementation details internal to have more freedom.

Reply:
Sure, I can make them private then. But note, if you decide to change the encoding, then you further complicate the Get/Set APIs on `nodeDB` because now you have to handle these cases.

Reply:
Yup, but at least we have the option. I've been burnt way too many times on exposing too large of an API surface and not being able to put the genie back in the bottle - but maybe it's just PTSD, used to work on SaaS APIs which were set in stone since things would immediately break out in the world if we changed anything at all.

Reply:
Great points!